pkg/monitor/cluster/cluster.go

```diff
@@ -137,18 +137,24 @@ func (mon *Monitor) Monitor(ctx context.Context) (errs []error) {
 		})
 	}
 
-	// If API Server is unreachable, don't need to run the next checks
-	statusCode, err := mon.getAPIServerPingCode(ctx)
-	if err != nil || statusCode != http.StatusOK {
+	//this API server healthz check must be first, our geneva monitor relies on this metric to always be emitted.
+	statusCode, err := mon.emitAPIServerHealthzCode(ctx)
+	if err != nil {
 		errs = append(errs, err)
-		friendlyFuncName := steps.FriendlyName(mon.getAPIServerPingCode)
+		friendlyFuncName := steps.FriendlyName(mon.emitAPIServerHealthzCode)
 		mon.log.Printf("%s: %s", friendlyFuncName, err)
 		mon.emitGauge("monitor.clustererrors", 1, map[string]string{"monitor": friendlyFuncName})
+	}
+	// If API is not returning 200, fallback to checking ping and short circuit the rest of the checks
+	if statusCode != http.StatusOK {
+		err := mon.emitAPIServerPingCode(ctx)
```

Comment:
Could we additionally emit  gauge metrics for:

1. Status code for ping
2. If `emitAPIServerPingCode` returned an err

This makes it easier to set up monitors for resource health check.

Reply:
1. `emitAPIServerPingCode` emits the statuscode
2. I can add a metric if this call fails